sdlib/d/gc/tcache.d

```diff
@@ -492,6 +492,25 @@ private:
 		triggerDeallocationEvent(npages * PageSize);
 	}
 
+	/**
+	 * Free directly to the arena, skip all threadCache metadata/caches.
+	 * Safe to call from a signal handler.
+	 */
+	void freeToArena(void* ptr) {
```

Comment:
It does not seems appropriate to me that this live in the thread cache, as the whole point of this thing it to NOT use the thread cache.

Reply:
OK, I moved it to the module where it's used (dmd/thread.d). I don't know that it belongs there, and I kind of like the function as an option (e.g. if you wanted to test frees without using the cache, this is a much better option than flushing the cache each free). But it can live there for now.